controllers/bookController.js

```diff
@@ -12,14 +17,46 @@ exports.addBook = catchAsync(async (req, res, next) => {
   });
 });
 exports.getAllBooks = catchAsync(async (req, res, next) => {
-  const books = await Book.find();
+  const queryObj = { ...req.query };
+  const excludedFields = ["page", "sort", "limit"];
+
+  // Exclude fields that are not for filtering
+  excludedFields.forEach((el) => delete queryObj[el]);
+
+  // Advanced filtering
+  let queryStr = JSON.stringify(queryObj);
+  queryStr = queryStr.replace(/\b(gte|gt|lte|lt)\b/g, (match) => `$${match}`);
+
+  // Initialize the query with filtering
+  let query = Book.find(JSON.parse(queryStr));
+
+  // Pagination
+  const page = req.query.page * 1 || 1;
```

Comment:
What's the essence of multiplying by 1?

Reply:
i am just making converting to number incase user provides a string type